tests/test_rocm_sanity.py

```diff
@@ -51,21 +58,36 @@ def test_rocm_output(self, rocm_info_output, to_search):
 
     def test_hip_printf(self):
         # Compiling .cpp file using hipcc
+        hipcc_executable = "./hipcc" if PLATFORM == "linux" else "hipcc.exe"
+        hipcc_check_executable = (
+            "hipcc_check" if PLATFORM == "linux" else "hipcc_check.exe"
+        )
         run_command(
             [
-                "./hipcc",
+                hipcc_executable,
                 str(THIS_DIR / "hipcc_check.cpp"),
                 "-o",
-                str(THIS_DIR / "hipcc_check"),
+                hipcc_check_executable,
             ],
             cwd=str(THEROCK_BIN_DIR),
         )
 
         # Running and checking the executable
-        process = run_command(["./hipcc_check"], cwd=str(THIS_DIR))
-        check.equal(process.returncode, 0)
-        check.greater(os.path.getsize(str(THIS_DIR / "hipcc_check")), 0)
+        hipcc_output_exec = (
+            "./" + hipcc_check_executable
+            if PLATFORM == "linux"
+            else hipcc_check_executable
+        )
```

Comment:
This platform handling is making the test quite hard to read. Does `executable` work on Linux with `cwd` specified, or does it really need to be `./executable`?